bedshift/bedshift.py

```diff
@@ -247,16 +252,24 @@ def shift_from_file(self, fp, shiftrate, shiftmean, shiftstdev, delimiter="\t"):
         num_shift = int(rows * shiftrate)
 
         intersect_regions = self._find_overlap(fp)
-        interlen = len(intersect_regions)
+        original_colnames = self.bed.columns
+        intersect_regions.columns = [str(col) for col in intersect_regions.columns]
+        self.bed.columns = [str(col) for col in self.bed.columns]
+        indices_of_overlap_regions = self.bed.reset_index().merge(intersect_regions)['index']
+        self.bed.columns = [int(col) for col in self.bed.columns]
```

Comment:
Why is this list comprehension needed?

Reply:
Otherwise, my version of pandas (1.2.3) complains about the merge operation, caused by non-string column names. So, we first convert the column names to string, and convert them back to the original type after merge, which is int.